url-sampler/src/main/java/io/opentelemetry/contrib/samplers/RuleBasedRoutingSampler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.contrib.samplers;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This sampler accepts a list of {@link SamplingRule}s and tries to match every proposed spans against those rules.
+ * Every rule describes a span's attribute, a pattern against which to match attribute's value, and a sampler that
+ * will make a decision about given span if match was successful.
+ * <p>
+ * If none of the rules matched, the default delegate sampler will make a decision.
+ */
+public class RuleBasedRoutingSampler implements Sampler {
+  private final List<SamplingRule> rules;
+  private final SpanKind kind;
+  private final Sampler delegate;
+
+  public RuleBasedRoutingSampler(List<SamplingRule> rules, SpanKind kind, Sampler defaultDelegate) {
```

Comment:
Let's make this package private and add `builder(SpanKind, Sampler)` static method

Reply:
Why do we want to hide this constructor? There is no actual requirement to use builder, isn't there?